wheatley/rhythm/abstract_rhythm.py

```diff
@@ -63,6 +63,6 @@ def initialise_line(
     ) -> None:
         """Allow the Rhythm object to initialise itself when 'Look to' is called."""
 
-    def sleep(self, seconds: float) -> None:  #  pylint: disable=no-self-use
```

Comment:
So close...
3.6 with the earlier version of pylint needs this, while the later pylint warns about the old option...

> python -m pylint wheatley
************* Module wheatley.row_generation.row_generator
wheatley/row_generation/row_generator.py:82:4: R0201: Method could be a function (no-self-use)
wheatley/row_generation/row_generator.py:87:4: R0201: Method could be a function (no-self-use)
************* Module wheatley.rhythm.abstract_rhythm
wheatley/rhythm/abstract_rhythm.py:66:4: R0201: Method could be a function (no-self-use)

Reply:
It's ✅ 🎉 

Sorry lots of small changes to review @kneasle, but I don't think anything functional
- formatting
- replace a couple of raw exceptions
- replace implicit optional types